sdks/python/apache_beam/runners/interactive/dataproc/dataproc_cluster_manager.py

```diff
@@ -0,0 +1,154 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+import logging
+
+try:
+  from google.cloud import dataproc_v1
+except ImportError:
+  raise ImportError(
+      'Google Cloud Dataproc not supported for this execution environment.')
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DataprocClusterManager:
+  """The DataprocClusterManager object simplifies the operations
+  required for creating and deleting Dataproc clusters for use
+  under Interactive Beam.
+  """
+  DEFAULT_NAME = 'interactive-beam-cluster'
+
+  def __init__(self, project_id=None, region=None, cluster_name=None):
+    """Initializes the DataprocClusterManager with properties required
+    to interface with the Dataproc ClusterControllerClient.
+    """
+
+    self.project_id = project_id
+    if region == 'global':
+      # The global region is unsupported as it will be eventually deprecated.
+      raise ValueError('Clusters in the global region are not supported.')
+    elif region:
+      self.region = region
+    else:
+      _LOGGER.warning(
+          'No region information was detected, defaulting Dataproc cluster '
+          'region to: us-central1.')
+      self.region = 'us-central1'
+
+    if cluster_name:
+      _LOGGER.warning(
+          'A user-specified cluster_name has been detected. '
+          'Please note that you will have to manually delete the Dataproc '
+          'cluster that will be created under the name: %s',
+          cluster_name)
+      self.cluster_name = cluster_name
+    else:
+      self.cluster_name = self.DEFAULT_NAME
+
+    self.cluster_client = dataproc_v1.ClusterControllerClient(
+        client_options={
+            'api_endpoint': f'{self.region}-dataproc.googleapis.com:443'
+        })
+
+  def _create_cluster(self, cluster):
+    """Attempts to create a cluster using attributes that were
+    initialized with the DataprocClusterManager instance.
+
+    Args:
+      cluster: Dictionary representing Dataproc cluster
```

Comment:
If possible, could you please also add the link to the schema of this dictionary?

Reply:
Yes, that would be a good idea. Added!